php-frontend/src/main/java/org/sonar/php/metrics/CommentLineVisitor.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010 SonarSource and Akram Ben Aissi
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.php.metrics;
+
+import com.google.common.collect.Sets;
+import org.sonar.plugins.php.api.tree.CompilationUnitTree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxTrivia;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+import java.util.Set;
+
+public class CommentLineVisitor extends PHPVisitorCheck {
+
+  private final boolean ignoreHeaderComments;
+  private Set<Integer> comments = Sets.newHashSet();
+  private Set<Integer> noSonarLines = Sets.newHashSet();
+
+  // seenFirstToken is required to track header comments (header comments are saved as trivias of first non-trivia token)
+  private boolean seenFirstToken = false;
+
+  public CommentLineVisitor(CompilationUnitTree tree) {
+    // FIXME SONARPHP-575
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)